src/main/java/com/byclosure/webcat/stepdefinitions/BrowsingSteps.java

```diff
@@ -0,0 +1,58 @@
+package com.byclosure.webcat.stepdefinitions;
+
+import com.byclosure.webcat.context.IContext;
+import com.byclosure.webcat.stepdefinitions.helpers.SeleniumInteractions;
+import com.google.inject.Inject;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import org.apache.commons.validator.routines.UrlValidator;
+import org.junit.Assert;
+import org.openqa.selenium.WebDriver;
+
+import java.io.UnsupportedEncodingException;
+
+public class BrowsingSteps {
+    
+    private final WebDriver driver;
+    private final SeleniumInteractions seleniumInteractions;
+
+    @Inject
+    public BrowsingSteps(WebDriver driver, IContext context){
+        this.driver = driver;
+        this.seleniumInteractions = new SeleniumInteractions(driver, context);
+    }
+    
+    @Given("^(?:|I )am on (.+)$")
+    @When("^(?:|I )go to (.+)$")
+    public void iAmOn(String url) {
```

Comment:
Use \"([^\"]*)\" instead of (.+)

for the api method names (#iAmOn in this case) do not focus in names that are used in the gherkin layer. This method should be called #goToUrl, #goTo, or #navigateTo.